rootfs/api/models/config.py

```diff
@@ -105,6 +111,40 @@ def set_tags(self, previous_config):
 
         raise DeisException(message)
 
+    def set_healthcheck(self, previous_config):
+        data = getattr(previous_config, 'healthcheck', {}).copy()
+        new_data = getattr(self, 'healthcheck', {}).copy()
+        # update the config data for healthcheck if they are not
+        # present for per proctype
+        if 'livenessProbe' in data.keys() or 'readinessProbe' in data.keys():
```

Comment:
this code is present to provide backward compatibility for the users who already set using previous versions of controller and are updating the controller with existing database.